File: LeetCode/C++/300-399/344.cpp
#include <iostream>
#include <vector>
using namespace std;
 
class Solution {
public:
    void reverseString(vector<char>& s) {
        for(int i = 0 ; i < s.size() / 2;++i){
            char temp = s[i];
            s[i] = s[s.size()-i-1];
            s[s.size()-i-1] = temp;
        }
    }
};
 
int main() {
	vector<char> s{'h','e','l','l','o'};
	Solution sol;
	sol.reverseString(s);
	for(char c : s){
		cout << c << " ";
	}
	return 0;
}

File: LeetCode/C#/300-399/344.cs
using System;

namespace LeetCode344
{
	public class Solution {
	    public void ReverseString(char[] s) {
	        int mid = s.Length / 2;
	        for(int i = 0 ; i < s.Length / 2;++i){
	            char a = s[i];
	            s[i] = s[s.Length - i - 1];
	            s[s.Length-i-1] = a;
	        }
	    }
	}
	
	public class Program
	{
		public static void Main()
		{
			Solution sol = new Solution();
			char[] s = new char[]{'h', 'e', 'l', 'l', 'o'};
			sol.ReverseString(s);
			foreach(char c in s)
			{
				Console.Write(c + " ");
			}
			
			Console.Read();
		}
	}
}
